octave/run_tnewamp1.m.in

```diff
@@ -1,5 +1,5 @@
 % Needed this to keep Travis happy, I guess on that version of
 % Linux/Octave, octave functions can't be invoked from the command
 % line, just scripts
-
-tnewamp1("../build_linux/src/cq_ref")
+addpath("@CMAKE_CURRENT_SOURCE_DIR@/octave")
+tnewamp1("@CMAKE_CURRENT_BINARY_DIR@/src/cq_ref")
```

Comment:
@tmiw so this picks up the CMAKE_CURRENT_BINARY_DIR environment variable?  I haven't seen this in Octave before :smile:

Reply:
@drowe67, as @hobbes1069 mentioned, this file is used to generate the .m file in the build folder. CMake will replace anything between the at signs with the value of that environment variable when generating that file.

Reply:
@hobbes1069 @tmiw I don't seem to have a copy of the .m files in my build folder, after the cmake/make/or ctest steps.  Is that an option I am missing?

Reply:
Are you sure? From my build folder:
```
$ find /home/build/build/codec2 -name "*.m"
/home/build/build/codec2/octave/run_tnewamp1.m
/home/build/build/codec2/octave/tnewamp1.m
```

Reply:
Here is what I just tried:
```
cd ~/tmp
git clone https://github.com/tmiw/codec2.git
cd codec2 && make build_linux && cmake .. && make && ctest
<40/40 tests pass>
find . -name "*.m"
<silence>
```

Trying a different build dir:
```
cd ~/tmp && mkdir build && cd build
cmake ~/tmp/codec2/ && make && ctest
<1 fail>
find . -name "*.m"
<nada>
pwd
/home/david/tmp/build
```

```
$cmake --version
cmake version 3.10.2
```

Reply:
@drowe67, you'll want to check out the osx-ut-fixes branch, not master.

Reply:
@tmiw OK so the changes to octave files and the ```.m.in``` naming convention appears to be a problem for my normal workflow where I run tests from inside codec2/octave:
```
cd codec2/octave
octave -- no-gui
ocatve> run_tnewamp1
````
Could we pass paths through an environment variable?  I think I do that in some other tests.

Reply:
@drowe67, there's this inside CMakeLists.txt:

```
    add_test(NAME test_OFDM_modem_octave_port
             COMMAND sh -c "PATH_TO_TOFDM=${CMAKE_CURRENT_BINARY_DIR}/unittest/tofdm octave --no-gui -qf ${CMAKE_CURRENT_SOURCE_DIR}/octave/tofdm.m"
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/octave)
             set_tests_properties(test_OFDM_modem_octave_port PROPERTIES
             PASS_REGULAR_EXPRESSION "fails: 0")
```

Was that what you meant? I suppose it's possible as long as there's a way to specify a default value for said environment variable so that it normally doesn't need to be set for your workflow.

Reply:
@tmiw is it OK if I task a look at those two tests that are in ```.m.in``` form?  I have some ideas to get them running without modification.

Reply:
@drowe67, go for it.